### DISCUSSION TITLE
How are retail vs. classic releases differentiated for addons hosted on github?

### DISCUSSION THREAD

**Discussion by ciphersimian** [layday/instawow#115] [Q&A]
I recently had to pin this addon in instawow because it started installing a newly posted release (v3.4.0) intended for classic WotLK but I'm on retail:

```
$ ./instawow -p wow1 rollback github:sfmict/hidingbar
- Select version of github:sfmict/hidingbar for rollback
 » v3.4.0
   - v9.2.12 (installed version)
   v9.2.11
```

I'm not sure if the issue is with instawow (or perhaps more likely) that the classic release of this addon either isn't supposed to be in the same github repo, or isn't tagged properly as being classic vs. retail?

https://github.com/sfmict/HidingBar/releases

**Comment by layday** [comment]
instawow should reject v3.4.0 and backtrack to find a matching release.  I'm also on retail and I get v9.2.12 on my end:

```
$ instawow install github:sfmict/hidingbar
✓ github:sfmict/hidingbar
  installed v9.2.12
```

Can you confirm that this is happening with the latest version of instawow?

**Reply by ciphersimian** [reply]
You're right, with the current version it's working as expected:
```
$ ./instawow -p wow1 remove github:sfmict/hidingbar
✓ github:sfmict/hidingbar
  removed
$ ./instawow -p wow1 install github:sfmict/hidingbar
✓ github:sfmict/hidingbar
  installed v9.2.12
$ ./instawow -p wow1 update
$ ./instawow --version
instawow, version 1.42.0
$
```

I was using `instawow-cli-1.40.0.post7.dev0-ubuntu-latest` when that v3.4.0 release came out (~6 days ago) and that's when I did a rollback after it mistakenly installed v3.4.0.

**Reply by layday** [reply]
Well, it was supposed to work like that in 1.40.0 too, not sure why it was picking up v3.4.0 😛 

P.S. `instawow rollback --undo github:sfmict/hidingbar` will unpin the add-on and install the latest version.

**Reply by layday** [reply]
Oh right, of course, it's because Wrath TOC versions were considered to be part of retail. The latest release added support for Wrath.
